browser_handler.go

```diff
@@ -217,59 +232,73 @@ func setContentType(compliance ComplianceOptions, h http.Header) {
 	if compliance.QualifyNamespaceDisabled {
 		h.Set("Content-Type", mime.TypeByExtension(".json"))
 	} else {
-		h.Set("Content-Type", YangDataJsonMimeType)
+		h.Set("Content-Type", string(YangDataJsonMimeType1))
 	}
 }
 
-func sendActionOutput(compliance ComplianceOptions, out io.Writer, output *node.Selection, a *meta.Rpc) error {
+func sendActionOutput(acceptType MimeType, compliance ComplianceOptions, wireFormat wireFormat, out io.Writer, output *node.Selection, a *meta.Rpc) error {
 	if !compliance.DisableActionWrapper {
 		// IETF formated output
 		// https://datatracker.ietf.org/doc/html/rfc8040#section-3.6.2
-		mod := meta.OriginalModule(a).Ident()
-		if _, err := fmt.Fprintf(out, `{"%s:output":`, mod); err != nil {
+		mod := meta.OriginalModule(a)
+		if _, err := wireFormat.writeRpcOutputStart(out, mod); err != nil {
 			return err
 		}
 	}
-	err := output.InsertInto(jsonWtr(compliance, out))
+	err := output.InsertInto(nodeWtr(acceptType, compliance, out))
 
 	if !compliance.DisableActionWrapper {
-		if _, err := fmt.Fprintf(out, "}"); err != nil {
+		if _, err := wireFormat.writeRpcOutputEnd(out); err != nil {
 			return err
 		}
 	}
 	return err
 }
 
-func jsonWtr(compliance ComplianceOptions, out io.Writer) node.Node {
+func nodeWtr(mime MimeType, compliance ComplianceOptions, out io.Writer) node.Node {
+	if mime.IsXml() {
+		wtr := &nodeutil.XMLWtr{
+			Out: out,
+		}
+		return wtr.Node()
+	}
 	wtr := &nodeutil.JSONWtr{
 		Out:              out,
 		QualifyNamespace: !compliance.QualifyNamespaceDisabled,
 	}
 	return wtr.Node()
 }
 
-func readInput(compliance ComplianceOptions, r *http.Request, a *meta.Rpc) (node.Node, error) {
+func nodeRdr(mime MimeType, in io.Reader) (node.Node, error) {
+	if mime.IsXml() {
+		return nodeutil.ReadXML(in)
+	}
+	return nodeutil.ReadJSONIO(in), nil
+}
+
+func readInput(compliance ComplianceOptions, contentType MimeType, r *http.Request, a *meta.Rpc) (node.Node, error) {
 	// not part of spec, custom feature to allow for form uploads
 	if isMultiPartForm(r.Header) {
 		return formNode(r)
-	} else if compliance.DisableActionWrapper {
-		return nodeutil.ReadJSONIO(r.Body), nil
 	}
-
+	n, err := nodeRdr(contentType, r.Body)
+	if err != nil {
+		return nil, err
+	}
+	if compliance.DisableActionWrapper {
+		return n, nil
+	}
+	m := meta.OriginalModule(a)
```

Comment:
With XML and strict compliance findNodeOutsideSchema(m, "input", n) will fail since the root element `input` is not in `xml_rdr::XmlNode::Nodes`.